### PR TITLE
Adjust mentions about CI to GitHub Actions

### PR DESCRIPTION
**Pull request by JohnTitor**
So, we've migrated our CI to GitHub Actions almost, except for macOS.
There're some mentions to Pipelines and even Travis in the guide, let's replace them with GHA. I don't think we have to mention macOS build but any suggestions are welcome.

### CODE REVIEW

File: src/tests/intro.md [hunk 1]
```diff
@@ -116,27 +116,29 @@ including:
...
 branch. When a PR is approved, it goes into a [queue] where merges are tested
-one at a time on a wide set of platforms using Azure Pipelines (currently over
-50 different configurations). Most platforms only run the build steps, some run
-a restricted set of tests, only a subset run the full suite of tests (see
-Rust's [platform tiers]).
-
-[Azure Pipelines]: https://dev.azure.com/rust-lang/rust/
+one at a time on a wide set of platforms using GitHub Actions (currently over
+50 different configurations). Due to the limit on the number of parallel jobs,
+we run CI under the [rust-lang-ci] organization except for PRs.
```
Comment (JohnTitor) [inline src/tests/intro.md:128]: > Due to the limit on the number of parallel jobs

Is this correct? @Mark-Simulacrum

File: README.md [hunk 2]
```diff
@@ -103,13 +103,16 @@ There is not a workaround for this error at the moment.  Comment out the `[outpu
 
 ## How to fix toolstate failures
 
+> **NOTE**: Currently, we do not track the toolstate due to the spurious failure,
+but we leave this for when we do it again in the future.
```
Comment (jyn514) [inline README.md:107]: What spurious failure is this referring to?

Comment (JohnTitor) [reply README.md:107]: I think this PR has detail explanation: https://github.com/rust-lang/rust/pull/71731

Comment (JohnTitor) [reply README.md:107]: So, it makes more sense to add this link here?

Comment (jyn514) [reply README.md:107]: Yes, I think so. Otherwise the first question is 'why is it turned off?' and there's not enough info here to figure it out.

Comment (jyn514) [reply README.md:107]: Also the wording is ambiguous and makes it look like _all_ toolstate is untracked, not just for rustc-dev-guide.

Comment (JohnTitor) [reply README.md]: Addressed via 7e24fae. Does it make you happy? :)

Comment (jyn514) [reply README.md]: Looks great :)

**Pull request merged**